test/unit/BPool/BPool_Unbind.tree

```diff
@@ -0,0 +1,25 @@
+BPool::Unbind
+├── when reentrancy lock is set
+│   └── it should revert
+├── when caller is NOT controller
+│   └── it should revert
+└── when caller is controller
+    ├── when token is not bound
+    │   └── it should revert
+    └── when token is bound
+        ├── when pool is finalized
+        │   └── it should revert
```

Comment:
in `bind` tree you're doing the `when pool is not finalized` as a direct leaf of the `when caller is controller`, and in `unbind` tree it is a dependant leaf of `when token is bound`..

for consistency, in `unbind` i'd do:
```
    ├── when token is not bound
    │   └── it should revert
    ├── when pool is finalized
    │   └── it should revert
    └── when token can be unbounded
        ├── when token is last on the tokens array
        │       └─ ...
        ├── when token is NOT last on the tokens array
```

Reply:
hmm, there's a catch, we have the bound checks before the finalized checks.

in `bind` it doesn't matter, because the pre-condition that "token needs NOT to be bound" is fulfilled by default, but in `unbind` we need to set it (since by default is non-bound)..

i don't know if branching the tree because of this is ok, is inconsistent at least.